### DISCUSSION TITLE
Move groups.io to GitHub discussions

### DISCUSSION THREAD

**Discussion by jlixfeld** [ib-api-reloaded/ib_async#2] [General]
The maintainers of the groups.io TWS API group who offered to host us as a sub-group have suggested that we may want to consider other options.

I tend to agree with their sentiments.

https://groups.io/g/twsapi/message/52573

I know there is Mel's discord and I'm happy to see that as an avenue for support and discussion, but I think that Discussions gives more flexibility, keeps things in one place for easy R&D and history, and all of the benefits that go along with that.  

Thoughts?

----

Regarding the subgroup issue, Jürgen and I have been discussing this further, and we wonder whether you have considered other options for support, particularly given that a subgroup is really a bit of a kludge and will almost certainly cause administrative problems for us because people will post to the wrong place, and especially since none of the original insync group posts will be carried forward.
 
As Jürgen succinctly expressed it in a private email to me (I hope he won’t mind me quoting it):
 
·         it is now clear, that this will not just be insync restored. None of the old content and none of the old members.
·         It is a complete restart and needs to be rebuilt and promoted from scratch
·         But as a sub-group needs to fit into our group and needs to basically follow our rules
 
I am not against it, but they need to know that they won't be as free as if they had their own community.
 
One option that is certainly worth considering is to simply do support via the GitHub facilities for the repositories. Users can raise Issues, create discussion documents and so on. Issues are notified via email to interested users. There are excellent formatting facilities via Markdown (which may be a learning curve for some, but the simplest usages are provided via buttons on the editing toolbar). Users would of course have to sign up with GitHub if they haven’t already done so.
 
I think this option has a lot to recommend it. Others, such as Discord, are also viable candidates.
 
As Jürgen said, we are still prepared to offer a subgroup if that’s really what you want. But please be sure!

----

**Comment by ClimberMel** [comment]
This may end up requiring multiple solutions as there is almost never a one size fits all for anything...

I use GitHub daily so I'm fine with it, but even though it has been suggested to use Mattsta's GH discussions (https://github.com/mattsta/ib_insync/discussions) as a forum, I personally feel it would become a mess really quickly.  A lot of people on the forums aren't always wanting to have a GitHub account and may not be good at following any sort of structure for creating separate discussions and I don't believe GH has the ability of automatically creating threads.

I have no problem using Discord obviously as I run a few servers on there, but as some have pointed out, they are harder to locate and again they don't "thread" messages, although they are nice at having "channels" which allow for setting topic areas.  Again there is no way to prevent people from posting in the wrong channel.  My discord for ib_insyc(https://discord.gg/WPEq37qZ)  was set up a long time ago for collaboration on code using ib_insync, so if people wanted to use discord as the new forum they would need to set up a new server as I prefer to try and keep mine focused on sharing code and sharing knowledge, but I'm not at Ewald's level of expertise where I want hundreds of questions on how do I do this...

I do like the email summaries that you can set up on groups.io and automatic threading of messages.  It also seems much simpler to administrate than other platforms... I also manage a groups.io that I inherited and the ability to screen new users does help keep the spammers out.

So my thoughts would be to use a sub group of twsapi as a general forum that people can find and use.  Of course it would follow the rules of twsapi, but it would also be handy as some are just discovering that twsapi can often do what they need without ib_insync.  In that sub forum we could have pinned messages that point to the GitHub with an explanation of reporting issues and joining discussions there or volunteering to help with code changes.  We could also have a link to any Discord server if that is what people want to use.

**Comment by mattsta** [comment]
Yeah, github usage isn't the absolute best for all of those use cases, but it seems to work for most other projects finding some balance between issues and discussions. This is an API library after all, so if people can't figure out github they may be beyond help anyway.

We can link to any live resources people want to maintain in the README, but questions about whether to continue or rebuild forum/chat are probably more about if anybody wants to continue running them. Happy to have them, but I don't have much time to be involved in those things currently.

**Comment by geepeem** [comment]
I agree with Matt - GitHub is as good a place as any to host a forum / discussions. We can always add on a new venue if we feel the need / can find volunteers to moderate. So I say let's keep it simple for now. 
BTW I just saw that there is already an issue with the new repo... 
"And so, it begins..." lol!

**Comment by mattsta** [comment]
Somebody also mentioned there may be an archive of the original groups (or maybe just years of emails digests?). We could restore those to a github repo/site too if anybody wants to provide a place to download them.

**Comment by ClimberMel** [comment]
I'm getting the digests and I've written a program to parse them out.MelSent from my Galaxy
-------- Original message --------From: Matt Stancliff ***@***.***> Date: 2024-03-20  4:56 p.m.  (GMT-08:00) To: ib-api-reloaded/ib_async ***@***.***> Cc: ClimberMel ***@***.***>, Comment ***@***.***> Subject: Re: [ib-api-reloaded/ib_async] Move groups.io to GitHub discussions (Discussion #2) 
Somebody also mentioned there may be an archive of the original groups (or maybe just years of emails digests?). We could restore those to a github repo/site too if anybody wants to provide a place to download them.
